### ISSUE TITLE
Qn: Backup strategy with deduplication

### ISSUE THREAD

**Issue by srinisubramanian** [borgbackup/borg#2378]
Recently came across Borg and have started using it for backup.  Based on the usage documentation, have also added prune to remove older backups.  

But given that this is a deduplicating backup and only differences are being maintained, if backups are pruned how can one do a complete restore?  Also what would be a correct backup strategy using Borg?  The need of course (as with all other backup tools) is to be able to easily restore completely in case of a failure.  How does one combine full / hourly backup and prune to have a good backup strategy?

Use case: LVM snapshot backup of databases

**Comment by level323** [comment]
I think it's possible you're misunderstanding how borg's deduplication works (and by extension how deduplicating backup software in general works). 

Conceptually, each backup run should be regarded as a complete snapshot of the files backed up at that point in time. It just so happens that borg has 'smarts' that enables it to only store data 'chunks' that are common to more than one file only once.

Pruning a previous backup only causes to be removed from the backup repository chunks of files that are unique to that backup (that is, those chunks did not exist in any other backup). This means that all other backups remain fully restorable.

In other words, pruning a set of prior backups from the repository does not affect in any way the ability to completely restore any/all of the backup sets that were not pruned (in your case the more recent backups).

Perhaps you are getting confused between incremental and deduplicating backup systems? It seems that your concerns may be based on a misunderstanding that borg is an incremental backup system. It is not.

**Comment by srinisubramanian** [comment]
Thanks @level323 for the detailed explanation. Indeed it was a confusion wrt incremental backup procedures.  This clarifies it all brilliantly.
